stackslib/src/util_lib/signed_structured_data.rs

```diff
@@ -0,0 +1,122 @@
+use clarity::vm::Value;
+use stacks_common::{
+    codec::StacksMessageCodec,
+    types::PrivateKey,
+    util::{
+        hash::{to_hex, Sha256Sum},
+        secp256k1::{MessageSignature, Secp256k1PrivateKey},
+    },
+};
+
+/// Message prefix for signed structured data. "SIP018" in ascii
+pub const STRUCTURED_DATA_PREFIX: [u8; 6] = [0x53, 0x49, 0x50, 0x30, 0x31, 0x38];
+
+pub fn structured_data_hash(value: Value) -> Sha256Sum {
+    let bytes = value.serialize_to_vec();
+    Sha256Sum::from_data(&bytes)
+}
+
+/// Generate a message hash for signing structured Clarity data.
+/// Reference [SIP018](https://github.com/stacksgov/sips/blob/main/sips/sip-018/sip-018-signed-structured-data.md) for more information.
+pub fn structured_data_message_hash(structured_data: Value, domain: Value) -> Sha256Sum {
```

Comment:
Why does this belong in `util_lib`?  Isn't this only used in testing?  If so, then please remove this file altogether and put this into `stackslib/src/chainstate/stacks/boot/mod.rs`.

Reply:
At the moment it's only used in tests, but we'll add it (in a separate PR) to `stacks-signer` so we can expose a CLI function like:

```
stacks-signer create-stacking-signature --stacker S --reward-cycle R
```

I don't have a strong opinion whether it should live here or not, and I'm not the best person to know. But I think it is helpful to expose somewhere that isn't just for tests.

Reply:
Sounds like this belongs in `libsigner` then.  `libsigner` can be a dev-dependency for `stackslib`.